_static/js/search.js

```diff
@@ -5,29 +5,34 @@ function hideEverything() {
     $("#helpmessage").hide();
 }
 
+
 window.addEventListener("load", function() {
     $('.ui.button').click(function() {
         $.getJSON("/api/check/" + encodeURIComponent($("input").val()), function(result) {
             if (result.result == 'verified') {
                 hideEverything();
+                $("#verifiedmessage").html($("input").val() + ' is a verified domain. You can trust the contents.');
+                $("#verifiedmessage").html($("#verifiedmessage").html());
                 $("#verified").css('display', 'flex');
             } else if (result.result == 'neutral') {
                 hideEverything();
+                $("#neutralmessage").html($("input").val() + ' wasn\'t recognized as a malicious domain, nor as verified domain. Be careful!');
+                $("#neutralmessage").html($("#neutralmessage").html());
                 $("#neutral").css('display', 'flex');
             } else if (result.result == 'blocked') {
                 hideEverything();
                 blocked = true;
                 var strLink = '';
                 if (result.type == 'domain' && 'category' in result.entries[0]) {
-                    $("#blacklistmessage").html('This domain was put on the blacklist for ' + result.entries[0].category.toLowerCase() + '.');
+                    $("#blacklistmessage").html($("input").val() + ' was put on the blacklist for ' + result.entries[0].category.toLowerCase() + '.');
                     strLink = '<a id="details" href="/scam/' + result.entries[0].id + '">Details <i class="chevron right small icon"></i></a>';
                 } else if(result.type == 'address') {
-					 $("#blacklistmessage").html('This address was put on the blacklist and is associated with '+ result.entries.length +' blocked domain(s)');
-					 strLink = '<a id="details" href="/address/' + result.input + '">Details <i class="chevron right small icon"></i></a>';
-				} else if(result.type == 'ip') {
-					 $("#blacklistmessage").html('This ip address was put on the blacklist and is associated with '+ result.entries.length +' blocked domain(s)');
-					 strLink = '<a id="details" href="/ip/' + result.input + '">Details <i class="chevron right small icon"></i></a>';
-				} 
+					          $("#blacklistmessage").html($("input").val() + ' was put on the blacklist and is associated with '+ result.entries.length +' blocked domain(s)');
+					          strLink = '<a id="details" href="/address/' + result.input + '">Details <i class="chevron right small icon"></i></a>';
+				        } else if(result.type == 'ip') {
+					          $("#blacklistmessage").html($("input").val() + ' was put on the blacklist and is associated with '+ result.entries.length +' blocked domain(s)');
```

Comment:
We need to add some sanitation here around `$("input").val()`. With the current implementation we can execute rouge JS - although not damaging in this application, we still need to sanitise.